app/scripts/services/serviceInstances.js

```diff
@@ -3,64 +3,105 @@
                                                DataService,
                                                NotificationsService) {
-
-    function deprovision(apiObject) {
+    var doDeprovision = function (apiObject) {
       var getErrorDetails = $filter('getErrorDetails');
+      var modalInstance;
+
       var modalScope = {
-        alerts: {
-          deprovision: {
-            type: 'error',
-            message: 'Service \'' + apiObject.spec.serviceClassName + '\' will be deleted and no longer available.'
-          }
-        },
-        detailsMarkup: 'Delete Service?',
+        kind: apiObject.kind,
+        displayName: apiObject.metadata.name,
         okButtonText: 'Delete',
         okButtonClass: 'btn-danger',
-        cancelButtonText: 'Cancel'
+        cancelButtonText: 'Cancel',
+        delete: function() {
+          modalInstance.close('delete');
+        }
       };
 
-      // TODO: we probably have to handle bindings in here.
-      // either:
-      // - automatically remove the bindings
-      // - tell the user they must manually unbind before continue
-      return $uibModal.open({
+      modalInstance =  $uibModal.open({
         animation: true,
-        templateUrl: 'views/modals/confirm.html',
+        templateUrl: 'views/modals/delete-resource.html',
         controller: 'ConfirmModalController',
         resolve: {
           modalConfig: function() {
             return modalScope;
           }
         }
-      })
-      .result.then(function() {
+      });
+
+      return modalInstance.result.then(function() {
         NotificationsService.hideNotification("deprovision-service-error");
 
-        return DataService.delete({
-            group: 'servicecatalog.k8s.io',
-            resource: 'serviceinstances'
-          },
-          apiObject.metadata.name,
-          { namespace: apiObject.metadata.namespace },
-          { propagationPolicy: null
-          }) // TODO - remove once this is resolved https://github.com/kubernetes-incubator/service-catalog/issues/942
-          .then(function() {
-            NotificationsService.addNotification({
-              type: "success",
-              message: "Successfully deleted provisioned service " + apiObject.metadata.name + "."
-            });
-          }, function(err) {
-            NotificationsService.addNotification({
-              id: "deprovision-service-error",
-              type: "error",
-              message: "An error occurred while deleting provisioned service " + apiObject.metadata.name + ".",
-              details: getErrorDetails(err)
-            });
+        var context = {namespace: apiObject.metadata.namespace};
+        var resource = {
+          group: 'servicecatalog.k8s.io',
+          resource: 'serviceinstances'
+        };
+
+        // TODO - remove once this is resolved https://github.com/kubernetes-incubator/service-catalog/issues/942
+        var opts = {
+          propagationPolicy: null
+        };
```

Comment:
Not related to this PR, but we should double-check that this isn't needed for bindings as well. Right now we're using `delete-link` for bindings and in some cases and not passing `propogationPolicy: null`

Reply:
Disregard, looks like we are handling it: https://github.com/openshift/origin-web-console/blob/master/app/scripts/directives/deleteLink.js#L153